lib/chef/azure/commands/enable.rb

```diff
@@ -169,7 +169,7 @@ def configure_chef_only_once
       end
       # Now the run chef-client with runlist in background, as we done want enable command to wait, else long running chef-client with runlist will timeout azure.
       puts "#{Time.now} Launching chef-client to register node with the runlist"
-      params = "-c #{bootstrap_directory}/client.rb -j #{bootstrap_directory}/first-boot.json -E _default -L #{@azure_plugin_log_location}/chef-client.log --once "
+      params = "-c #{bootstrap_directory}/client.rb -j #{bootstrap_directory}/first-boot.json -E #{config[:environment]} -L #{@azure_plugin_log_location}/chef-client.log --once "

```

Comment:
@siddheshwar-more handled the corner case but in a different way, so that config[:environment] attribute has the aappropriate value available for further use (if any).